pkg/handler/snapshots.go

```diff
@@ -206,3 +224,211 @@ func (sh *SnapshotHandler) getRepoConfigurationFile(c echo.Context) error {
+				errs[i] = err
+			}
+		}
+		if hasErr {
+			return ce.NewErrorResponseFromError("Error clearing snapshot deleted_at fields", errs...)
+		}
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(enqueueErr), "Error enqueueing task", enqueueErr.Error())
+	}
+
+	return c.NoContent(http.StatusNoContent)
+}
+
+func (sh *SnapshotHandler) enqueueDeleteSnapshotsTask(c echo.Context, orgID, repoUUID string, snapshotUUIDs ...string) error {
+	accountID, _ := getAccountIdOrgId(c)
+	payload := payloads.DeleteSnapshotsPayload{RepoUUID: repoUUID, SnapshotsUUIDs: snapshotUUIDs}
+	task := queue.Task{
+		Typename:   config.DeleteSnapshotsTask,
+		Payload:    payload,
+		OrgId:      orgID,
+		AccountId:  accountID,
+		ObjectUUID: utils.Ptr(repoUUID),
+		ObjectType: utils.Ptr(config.ObjectTypeRepository),
+		RequestID:  c.Response().Header().Get(config.HeaderRequestId),
+	}
+
+	taskID, err := sh.TaskClient.Enqueue(task)
+	if err != nil {
+		logger := tasks.LogForTask(taskID.String(), task.Typename, task.RequestID)
+		logger.Error().Msg("error enqueuing task")
+		return err
+	}
+
+	return nil
+}
+
+func (sh *SnapshotHandler) isDeleteInProgress(c echo.Context, orgID, repoUUID string) error {
+	inProgressTasks, err := sh.DaoRegistry.TaskInfo.
+		FetchActiveTasks(c.Request().Context(), orgID, repoUUID, config.DeleteRepositorySnapshotsTask, config.DeleteSnapshotsTask)
+
+	if err != nil {
+		return ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error fetching delete repository snapshots task", err.Error())
+	}
+	if len(inProgressTasks) >= 1 {
+		return ce.NewErrorResponse(http.StatusBadRequest, "Error deleting snapshot", "Delete is already in progress")
+	}
+
+	return nil
+}
+
+func (sh *SnapshotHandler) isDeleteAllowed(c echo.Context, orgID, repoUUID string, snapshotUUIDs ...string) ([]models.Snapshot, error) {
+	_, err := sh.DaoRegistry.RepositoryConfig.Fetch(c.Request().Context(), orgID, repoUUID)
+	if err != nil {
+		return nil, ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error fetching repository config", err.Error())
+	}
+
+	if len(snapshotUUIDs) == 0 {
+		return nil, ce.NewErrorResponse(http.StatusBadRequest, "Error deleting snapshots", "Request body must contain at least 1 snapshot UUID to delete.")
+	}
+	if BulkDeleteLimit < len(snapshotUUIDs) {
+		limitErrMsg := fmt.Sprintf("Cannot delete more than %d snapshots at once.", BulkDeleteLimit)
+		return nil, ce.NewErrorResponse(http.StatusRequestEntityTooLarge, "Error deleting repositories", limitErrMsg)
+	}
+
+	err = sh.isDeleteInProgress(c, orgID, repoUUID)
+	if err != nil {
+		return nil, err
+	}
+
+	repoSnaps, err := sh.DaoRegistry.Snapshot.FetchForRepoConfigUUID(c.Request().Context(), repoUUID)
+	if err != nil {
+		return nil, ce.NewErrorResponse(ce.HttpCodeForDaoError(err), "Error fetching snapshots", err.Error())
+	}
+	if len(repoSnaps) <= len(snapshotUUIDs) {
+		return nil, ce.NewErrorResponse(http.StatusBadRequest, "Error deleting snapshot", "Can't delete all the snapshots in the repository")
+	}
```

Comment:
Related to this, I'm seeing an issue:

1. I have a repo with 3 snapshots
2. I bulk delete 2 snapshots
3. I make the same request again, this time I see the error "Can't delete all the snapshots in the repository". I would expect to see something more like "snapshots not found".

Reply:
yeah, that might be a better order for the validation checks 👍🏼